web/src/upload.js

```diff
@@ -450,15 +478,48 @@ export function initialize() {
         }
     });
 
+    let drag_counter = 0;
     // Allow the app panel to receive drag/drop events.
-    $(".app, #navbar-fixed-container").on("dragover", (event) => event.preventDefault());
+    $(".app, #navbar-fixed-container").on("dragover", (event) => {
+        event.preventDefault();
+        // Show the upload overlay during drag events.
+        $(".upload-overlay-container").addClass("upload-overlay-visibility");
+        // Adjust z-index so that message box is visible.
+        if ($(".message_comp").css("display") === "block") {
+            $("#compose").css({"z-index": "10001"});
```

Comment:
It would probably be good to put each of the z-index values here into a variable, just to make future changes or refactoring easier (ideally, we'd just be able to toggle a class here and have a CSS file worry about what the `z-index` is on that class.